docs/index.bs

```diff
@@ -970,9 +971,10 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
         1. Let |promise| be [=a new promise=].
         1. Run the following steps [=in parallel=]:
             1. Let |registration| be [=this=]'s associated [=/service worker registration=].
-            1. If |registration|'s [=active worker=] is null, [=reject=] |promise| with an "{{InvalidStateError}}" {{DOMException}}, and abort these steps.
-            1. Set |registration|'s [=navigation preload enabled flag=].
-            1. Resolve |promise| with undefined.
+            1. [=Queue a task=] on |promise|'s [=relevant settings object=]'s [=responsible event loop=], using the [=DOM manipulation task source=], to run the following steps:
+                1. If |registration|'s [=active worker=] is null, [=reject=] |promise| with an "{{InvalidStateError}}" {{DOMException}}, and abort these steps.
+                1. Set |registration|'s [=navigation preload enabled flag=].
```

Comment:
If this is for protecting [=navigation preload enabled flag=] to be updated in parallel, I guess we also need the code to read [=navigation preload enabled flag=] in Handle Fetch to be executed with queue?

Reply:
In `Handle Fetch` algorithm I see [=navigation preload enabled flag=] is being used to check and also don't see they are running the steps are in [=in parallel=]

Reply:
Stepping back and see if it is possible to update [=navigation prelaod flag=] from arbitrary location.
Then, I totally got confused.

`ServiceWorkerRegistration` has `NavigationPreloadManager` as a readonly object.  Then, I assume it is not possible to call `enable()`/`disable()` because it updates a field inside `NavigationPreloadManager`.  If it is allowed, then I suppose it technically possible to execute `enable()`/`disable()` from arbitrary ServiceWorker clients?  I suppose each client is loaded and executed asynchronously, and we might need to assume the case that the flag is changed during `Handle Fetch`.

Stepping back again, `Handle Fetch` just read a boolean flag, I guess the operation here is atomic.  Then, it might be fine to read/write it at the same time.